torch/autograd/profiler.py

```diff
@@ -101,55 +101,49 @@ def export_chrome_trace(self, path):
         Arguments:
             path (str): Path where the trace will be written.
         """
-        import json
+        import os
         with open(path, 'w') as f:
             chrome_events = []
             next_id = 0
+            f.write("[")
             for evt in self:
-                chrome_events.append(dict(
-                    name=evt.name,
-                    ph='X',
-                    ts=evt.cpu_interval.start,
-                    dur=evt.cpu_interval.elapsed_us(),
-                    tid=evt.thread,
-                    pid='CPU functions',
-                    args={},
-                ))
+                f.write("{\"name\": \"" + str(evt.name) + "\", "
+                        + "\"ph\": \"X\", "
+                        + "\"ts\": " + str(evt.cpu_interval.start) + ", "
+                        + "\"dur\": " + str(evt.cpu_interval.elapsed_us()) + ", "
+                        + "\"tid\": " + str(evt.thread) + ", "
+                        + "\"pid\": \"CPU functions\", " + "\"args\": {}" + "}, ")
```

Comment:
Wouldn't it be more idiomatic Python to use `.format()` here? (Or is that even slower?)

Reply:
Let me try and get back to you.